code-challenges.test.js

```diff
@@ -23,30 +39,86 @@ const secretCodeWord3 = "Eccentric"
 
 // b) Create the function that makes the test pass.
 
+const codedString = (str) => {
+    let code = {"a":4, "e":3, "i":1, "o":0}
+    return str.split("").map((value) =>{
+        return(object.keys(code).includes(value.toLowercase())) ? code[value.toLowercase()] : value
+    }).join("")
+}
+
+// pseudo code: 
+// input: function that takes in a string
+// output/return: a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
+// I created a function that took in a string and returned an array of strings with the split "" and used the .map to iterate through every element in the array. Then i ask for a return and used the .keys method which returns an array of the objects strings. i also used the .includes() method that looks for one string that could be in another string returning the true or false as it should. And the .toLowerCase converts a string to lower case letters
+
```

Comment:
Just for clarity `.keys` returns an array of the keys in an object, which for this object happens to be strings.